substrate/utils/frame/rpc/support/src/lib.rs

```diff
@@ -140,23 +144,46 @@ pub struct StorageQuery<V> {
 	_spook: PhantomData<V>,
 }
 
-impl<V: FullCodec> StorageQuery<V> {
+impl<V: FullCodec + 'static> StorageQuery<V> {
 	/// Create a storage query for a StorageValue.
-	pub fn value<St: StorageValue<V>>() -> Self {
-		Self { key: StorageKey(St::storage_value_final_key().to_vec()), _spook: PhantomData }
+	pub fn value<P>() -> Self
```

Comment:
instead of `<P>` it should be `<St: StorageValue<P, V, Q, E>, Q: .., E: ...,>` and same for the other one, ideally the tests should be man

Reply:
Or better use the traits StorageValue so we don't even change the signature of the function and avoid the breaking change

Reply:
Thanks fo the review @gui1117 

Will implement the suggested changes and notify you for next review 🙏

Reply:
yes

Reply:
The idea is that the generic `P` is generated by the macro so it would be better if user used the storage type I think

Reply:
they will pass in the storage type that implements this StorageValue trait?

Reply:
yes